src/index.ts

```diff
@@ -14,255 +18,121 @@ const TrieSearch = require('trie-search');
- */
-function logError(error: string): void {
-    console.log('\x1b[31m%s\x1b[0m', `[${new Date().toLocaleString()}]: ${error}`);
-}
-
-/**
- * Returns random hex color
- */
-function generateRandomColor(): string {
-    let colors = ['#FBE8A6']; // choose set of colors that fit with the color scheme
-    return colors[Math.floor(Math.random() * colors.length)];
-}
 
 /**
  * Sends socket message with data to everyone in the room
- * If senderid is specified, only sender will not receive message.
+ * If senderId is specified, only sender will not receive message.
  *
- * @param roomid - room to send socket message to
+ * @param roomId - room to send socket message to
  * @param msg - socket message to send
  * @param data - data to send
- * @param senderid - if value is included, doesn't send to sender.
+ * @param senderId - if value is included, doesn't send to sender.
  */
-function sendSocketMsgToRoom(roomid: string, msg: string, data: any, senderid: string) {
-    const room = rooms[roomid];
+function sendSocketMsgToRoom(roomId: string, msg: string, data: any, senderId: string) {
+    const room = roomService.getRoom(roomId);
     let invitedUserIds = Object.keys(room.invited);
-    invitedUserIds.forEach(socketid => {
-        if (senderid !== socketid && room.invited[socketid].accepted) {
-            io.to(socketid).emit(msg, data);
+    invitedUserIds.forEach(socketId => {
+        if (senderId !== socketId && room.invited[socketId].accepted) {
+            io.to(socketId).emit(msg, data);
         }
     });
 }
 
-function handleLeaveRoom(roomid: string, userid: string) {
-    if (rooms[roomid]) {
-        const room = rooms[roomid];
-        // Remove from room
-        delete room.invited[userid];
-        logInfo(`${displayNames[userid]} disconnected from room ${roomid}`);
-
-        // Update users in current room or delete room if empty.
-        let invitedUserIds = Object.keys(room.invited);
-        if (invitedUserIds.length === 1) {
-            io.to(invitedUserIds[0]).emit(Constants.LEAVE_ROOM, { roomid: roomid });
-            delete rooms[roomid];
-            logInfo(`Everyone left room ${roomid} and it has been closed.`);
-        } else {
-            invitedUserIds.forEach(socketid => {
-                if (room.invited[socketid].accepted) {
-                    io.to(socketid).emit(Constants.ROOM_STATUS, {
-                        type: Constants.USER_DISCONNECT,
-                        roomid: roomid,
-                        invited: room.invited,
-                        full: invitedUserIds.every(userid => {
-                            return room.invited[userid].accepted;
-                        }),
-                        owner: room.owner,
-                        userid: userid,
-                    });
-                }
-            });
-        }
-    }
-}
-
 // Socket Connection
 
-const io = socketio();
-io.on('connection', client => {
+io.on('connection', (client: any) => {
```

Comment:
There's no type for `client`?

Reply:
Also, should we make `connection` a constant?

Reply:
Both of these can be part of another PR tho. Just wanna comment to keep track of things i notice now.

Reply:
Changed!